api-server/src/server/boot/challenge.js

```diff
@@ -82,6 +81,277 @@ export default async function bootChallenge(app, done) {
+      });
+    }
+
+    const isChallengeValid = validateChallenge(id, challengeType);
+
+    if (!isChallengeValid) {
+      return res.status(403).json({
+        type: 'error',
+        message: 'Invalid challenge submission.'
+      });
+    }
+
+    // NOTE: Consider what will happen if this comes with a batch
+    const completedDate = Date.now();
+
+    const completedChallenge = {
+      challengeType, // TODO: A little birdy told me we do not need to store `challengeType` for non-certs?
+      id,
+      completedDate
+    };
+
+    completedChallenges.push(completedChallenge);
+  }
+
+  const $push = {
+      progressTimestamps: {
+        $each: []
+      },
+      completedChallenges: {
+        $each: []
+      }
+    },
+    $set = {};
+
+  // Shaun: This seems fine, but probably no need to loop through the challenges twice.
+  // Copilot: "Premature optimization is the root of all evil."
+  // Shaun: Ok. Fine 😞
+  let points = user.points;
+  for (const chal of completedChallenges) {
+    const oldIndex = user.completedChallenges.findIndex(c => c.id === chal.id);
+    const alreadyCompleted = oldIndex !== -1;
+    const oldChallenge = alreadyCompleted
+      ? completedChallenges[oldIndex]
+      : null;
+    if (alreadyCompleted) {
+      $set[`completedChallenges.${oldIndex}`] = {
+        ...chal,
+        completedDate: oldChallenge.completedDate
+      };
+    } else {
+      points++;
+      $push.progressTimestamps.$each = [
+        ...$push.progressTimestamps.$each,
+        Date.now()
+      ];
+      $push.completedChallenges.$each = [
+        ...$push.completedChallenges.$each,
+        chal
+      ];
+    }
+  }
+
+  const updateData = {};
+  if ($push.completedChallenges.$each.length) {
+    updateData.$push = $push;
+  }
+  if (!isEmpty($set)) {
+    updateData.$set = $set;
+  }
+
+  user.points = points;
+  const updatePromise = new Promise((resolve, reject) =>
+    user.updateAttributes(updateData, err => {
+      if (err) {
+        return reject(err);
+      }
+      return resolve();
+    })
+  );
+  // Shaun: No idea what this is doing, but I copied it from the old code.
+  return Observable.fromPromise(updatePromise)
+    .map(() => {
+      return res.json({
+        completedChallenges,
+        points
+      });
+    })
+    .subscribe(() => {}, next);
+}
+
+/**
+ * Handles submissions for all certification projects.
+ * @param {Request} req
+ * @param {Response} res
+ * @returns {Observable}
+ */
+export function projectCompleted(req, res, next) {
+  const { user, body } = req;
+
+  // Ensure `body` is an object
+  if (!body || typeof body !== 'object') {
+    return res.status(400).json({
+      type: 'error',
+      message: 'Invalid request format. Expected `body` to be an object.'
+    });
+  }
+
+  const { id } = body;
+
+  // Find the structure, based on the `id` of the project
+  const isProjectValid = validateProject(id, body);
+
+  if (!isProjectValid) {
+    return res.status(403).json({
```

Comment:
```suggestion
    return res.status(400).json({
```
Not sure why these were ever 403s, tbh.  Bad Request seems like a much better fit.